build.gradle

```diff
@@ -142,19 +145,55 @@ tasks.withType(JavaCompile).configureEach {
 	options.compilerArgs.add '-XDstringConcat=inline'
 }
 
-build.dependsOn('installLocalGitHookPostCheckout')
-tasks.register('installLocalGitHookPostCheckout', Copy) {
-	group = 'git'
-	from new File(rootProject.rootDir, '.scripts/post-checkout')
-	into { new File(rootProject.rootDir, '.git/hooks') }
-	filePermissions {
-		unix(0775)
+
+// Setup local git hooks
+
+def ensureSymlink(link, target) {
+	if (!Files.isSymbolicLink(link)) {
+		Files.deleteIfExists(link)
+
+		Files.createSymbolicLink(link, target)
+
+		def relative = { f -> rootDir.toPath().relativize(f) }
```

Comment:
what is f?

Reply:
come on this closure is literally used twice ever and both of those are on the next line the whole idea is to refactor out the repetition so the template string can be cleaner. If I give everything long names I might as well just write the entire `relativize` in the template string

Reply:
this is a function that gets path and make it relative to root, i think the naming of the function and f should represent it

Reply:
you can call the func rootRelative and f path

Reply:
I called it `shortname` in the new code if you look, because that's the reason it's used here

Reply:
yes i saw, but the function still gets path and make it relative to root and names should say that